Source/WebCore/platform/graphics/Font.h

```diff
@@ -425,6 +431,62 @@ ALWAYS_INLINE FloatRect Font::boundsForGlyph(Glyph glyph) const
     return bounds;
 }
 
+#if USE(CORE_TEXT)
+ALWAYS_INLINE Vector<FloatRect> Font::boundsForGlyphs(std::span<const Glyph> glyphs) const
+{
+    ASSERT(glyphs.size());
+
+    Vector<FloatRect> glyphBounds(glyphs.size(), FloatRect());
+    if (glyphs.size() == 1) {
+        glyphBounds[0] = boundsForGlyph(glyphs[0]);
+        return glyphBounds;
+    }
+
+    Vector<Glyph> glyphsNeedingMeasurement;
+    glyphsNeedingMeasurement.reserveInitialCapacity(glyphs.size());
+
+    Vector<size_t> positionsNeedingMeasurement;
+    positionsNeedingMeasurement.reserveInitialCapacity(glyphs.size());
+
+    for (size_t glyphIndex = 0; glyphIndex < glyphs.size(); ++glyphIndex) {
+        auto glyph = glyphs[glyphIndex];
+        if (isZeroWidthSpaceGlyph(glyph))
+            continue;
+
+        if (m_glyphToBoundsMap) {
+            auto bounds = m_glyphToBoundsMap->metricsForGlyph(glyph);
+            if (bounds.width() != cGlyphSizeUnknown) {
+                glyphBounds[glyphIndex] = bounds;
+                continue;
+            }
+        }
+
+        glyphsNeedingMeasurement.append(glyph);
+        positionsNeedingMeasurement.append(glyphIndex);
+    }
+
+    if (glyphsNeedingMeasurement.isEmpty())
+        return glyphBounds;
+
+    if (!m_glyphToBoundsMap)
+        m_glyphToBoundsMap = makeUnique<GlyphMetricsMap<FloatRect>>();
+
+    const auto& measuredBounds = platformBoundsForGlyphs(glyphsNeedingMeasurement);
+
+    ASSERT(measuredBounds.size() == glyphsNeedingMeasurement.size());
+    size_t index = 0;
+    for (auto&& bounds : measuredBounds) {
```

Comment:
is it `auto&&` so you could try to `WTFMove` into `glyphBounds`? I wonder why that didn't work. Maybe just `auto&` would be enough to let you move it?